SingularityService/src/main/java/com/hubspot/singularity/resources/HistoryResource.java

```diff
@@ -107,9 +110,42 @@ public SingularityDeployHistory getDeploy(@ApiParam("Request ID for deploy") @Pa
   }
 
   @GET
+  @Path("/request/{requestId}/deploy/{deployId}/tasks/active")
+  @ApiOperation("Retrieve the task history for a specific deploy.")
+  public List<SingularityTaskIdHistory> getActiveDeployTasks(
+      @ApiParam("Request ID for deploy") @PathParam("requestId") String requestId,
+      @ApiParam("Deploy ID") @PathParam("deployId") String deployId) {
+
+    List<SingularityTaskId> taskIds = new ArrayList<SingularityTaskId>();
+    for (SingularityTaskId id : taskManager.getActiveTaskIdsForRequest(requestId)) {
+      if (id.getDeployId().equals(deployId)) {
+        taskIds.add(id);
+      }
+    }
+    return taskHistoryHelper.getHistoriesFor(taskIds);
```

Comment:
i'd move these lines into a new method in `TaskHistoryHelper`, perhaps `getTaskHistoryForDeploy()`